src/Reference/HasMany.php

```diff
@@ -53,11 +54,12 @@ protected function referenceOurValue(): Field
      */
     public function ref($defaults = []): Model
     {
-        return $this->getModel($defaults)
-            ->addCondition(
-                $this->their_field ?: ($this->owner->table . '_' . ($this->owner->id_field ?: 'id')),
-                $this->getOurValue()
-            );
+        $ourModel = $this->getOurModel();
+
+        return $this->getTheirModel($defaults)->addCondition(
+            $this->their_field ?: ($ourModel->table . '_' . ($ourModel->id_field ?: 'id')),
```

Comment:
we should have more like two protected methods for build and parse (I think related with table alias regexes above)

Reply:
Also here - feel free to implement your ideas in this PR

Reply:
related with https://github.com/atk4/data/pull/675#discussion_r460437912 , this should be probably solved in this PR

Reply:
This PR is about pure renaming. Let's merge this PR and you look into it when you have time and willingness for it.